libs/providers/launchdarkly-client/src/lib/launchdarkly-client-provider.ts

```diff
@@ -34,73 +38,110 @@ export class LaunchDarklyClientProvider implements Provider {
     name: 'launchdarkly-client-provider',
   };
 
+  private readonly ldOptions: LDOptions | undefined;
   private readonly logger: Logger;
+  private _client?: LDClient;
+
+  public events = new OpenFeatureEventEmitter();
+
+  /*
+   * implement status field/accessor
+   * https://openfeature.dev/specification/sections/providers#requirement-242
+   * */
+  private _status: ProviderStatus = ProviderStatus.NOT_READY;
 
-  constructor(private client: LDClient, options: LaunchDarklyProviderOptions = {}) {
-    if (options.logger) {
-      this.logger = options.logger;
+  set status(status: ProviderStatus) {
+    this._status = status;
+  }
+
+  get status() {
+    return this._status;
+  }
+
+  constructor(private readonly envKey: string, { logger, ...ldOptions }: LaunchDarklyProviderOptions) {
+    if (logger) {
+      this.logger = logger;
     } else {
       this.logger = basicLogger({ level: 'info' });
     }
+    this.ldOptions = { ...ldOptions, logger: this.logger };
+
+  }
+
+  private get client(): LDClient {
+    if (!this._client) {
+      throw new GeneralError('Provider is not initialized');
+    }
+    return this._client;
   }
 
-  async initialize(context: EvaluationContext): Promise<void> {
-    const newContext = this.translateContext(context);
-    const oldContext = this.client.getContext();
-    const ignoreAnonymous = newContext.anonymous && oldContext.anonymous && !newContext.key
-    if(!ignoreAnonymous && !isEqual(newContext, oldContext)) {
-       await this.client.identify(newContext);
+  async initialize(): Promise<void> {
+    /* for simplicity always initialize with anonymous by default*/
+
+    this._client = initialize(this.envKey, { anonymous: true }, this.ldOptions);
```

Comment:
I don't find initializing with an anonymous context, unconditionally, to be a good idea. It generates less than optimal traffic (possibly an extra stream, and always an extra poll). Additionally it may cause substantial confusion with bootstrapping, as you ideally want that data to be associated with the correct context.

Reply:
Would your recommended solution be to make this a required part of the configuration then?

Reply:
So, before, it allowed init with a context, and then needed to do a comparison. This was because the client existed before the provider.

Here I am confused though, I think this should just use the context that should be passed to `initialize`. Then the user would just need to set a global context before setting the provider? In which case it would just get initialized with that context?

Reply:
Yeap I think I focus on my use case which is a error obviously, fixing it now, thanks @kinyoklion